internal/proxy/search_util.go

```diff
@@ -141,51 +165,16 @@ func parseSearchInfo(searchParamsPair []*commonpb.KeyValuePair, schema *schemapb
 	}
 
 	// 5. parse group by field and group by size
-	groupByFieldName, err := funcutil.GetAttrByKeyFromRepeatedKV(GroupByFieldKey, searchParamsPair)
-	if err != nil {
-		groupByFieldName = ""
-	}
-	var groupByFieldId int64 = -1
-	if groupByFieldName != "" {
-		fields := schema.GetFields()
-		for _, field := range fields {
-			if field.GetNullable() {
-				return nil, 0, merr.WrapErrParameterInvalidMsg(fmt.Sprintf("groupBy field(%s) not support nullable == true", groupByFieldName))
-			}
-			if field.Name == groupByFieldName {
-				groupByFieldId = field.FieldID
-				break
-			}
-		}
-		if groupByFieldId == -1 {
-			return nil, 0, merr.WrapErrFieldNotFound(groupByFieldName, "groupBy field not found in schema")
-		}
-	}
-
-	var groupSize int64
-	groupSizeStr, err := funcutil.GetAttrByKeyFromRepeatedKV(GroupSizeKey, searchParamsPair)
-	if err != nil {
-		groupSize = 1
-	} else {
-		groupSize, err = strconv.ParseInt(groupSizeStr, 0, 64)
-		if err != nil || groupSize <= 0 {
-			groupSize = 1
-		}
-	}
-	if groupSize > Params.QuotaConfig.MaxGroupSize.GetAsInt64() {
-		return nil, 0, merr.WrapErrParameterInvalidMsg(
-			fmt.Sprintf("input group size:%d exceeds configured max group size:%d", groupSize, Params.QuotaConfig.MaxGroupSize.GetAsInt64()))
-	}
-
+	var groupByFieldId, groupSize int64
 	var groupStrictSize bool
-	groupStrictSizeStr, err := funcutil.GetAttrByKeyFromRepeatedKV(GroupStrictSize, searchParamsPair)
-	if err != nil {
-		groupStrictSize = false
+	if isAdvanced {
+		groupByFieldId, groupSize, groupStrictSize = rankParams.groupByFieldId, rankParams.groupSize, rankParams.groupStrictSize
```

Comment:
rankParams.GetXXX instead of rankParams.groupByFieldId

Reply:
changed